ea_airflow_util/callables/ssm.py

```diff
@@ -76,17 +83,34 @@ def refresh(self):
         self._keys = {}
         self._substores = {}
 
+        # Different logic is used when passing a complete prefix vs a wildcard.
+        if self.TENANT_REPR in self._prefix:
+            filter_prefix, _ = self._prefix.split(self.TENANT_REPR, 1)
+        else:
+            filter_prefix = self._prefix
+
         paginator = self._client.get_paginator('describe_parameters')
         pager = paginator.paginate(
             ParameterFilters=[
-                dict(Key="Path", Option="Recursive", Values=[self._prefix])
+                dict(Key="Path", Option="Recursive", Values=[filter_prefix])
             ]
         )
 
         for page in pager:
             for p in page['Parameters']:
-                paths = p['Name'][len(self._prefix):].split('/')
+
+                # If a wildcard is in the prefix string, extract and set the tenant code as the first path element.
+                if self.TENANT_REPR in self._prefix:
+                    inferred_tenant = re.search(self._prefix.replace(self.TENANT_REPR, "(.*)"), p['Name']).group(1)
```

Comment:
would there every be a case where no TENANT_REPR doesn't exist anywhere in p['Name']? If so, this would error, I think

also, this line 104 could use some more friendly commenting, but that's a nitpick :)

Reply:
Added more comments. I would like to come up with an example of the situation you describe so we can work around it.